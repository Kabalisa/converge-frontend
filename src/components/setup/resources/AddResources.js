import React from 'react';
import ActionButtons from '../../commons/ActionButtons';
import MrmModal from '../../commons/Modal';
import { Input } from '../../commons';
import '../../../assets/styles/addresource.scss';

/**
 * Add Resource Component
 *
 * @extends React.Component
 *
 * @returns {JSX}
 */
export class AddResource extends React.Component {
  state = {
    amenity: '',
    closeModal: false,
    resourceQuantity: 0,
  };

  resourceQuantity = React.createRef();
  amenity = React.createRef();

  /**
   * Ensures that the modal for adding a resource closes
   * when a user hits CANCEL on the modal or when the
   * adding a resource is successful
   *
   * @returns {void}
   */
  handleCloseModal = () => {
    this.setState({ closeModal: true });
  };

  /**
   * Ensures that the state is updated basing on the
   * user input
   *
   * @param {object} event The input event parameter
   * @param {number} capacity
   *
   * @returns {void}
   */
  handleInputChange = (event, capacity = 0) => {
    const { name, value } = event.target;
    this.setState({ [name]: value, resourceQuantity: capacity });
  };

  /**
   * It updates the state value of closeModal
   * to false whenever the modal closes
   *
   * @returns {void}
   */
  handleModalStateChange = () => {
    this.state.closeModal && this.setState({ closeModal: false });
  };

  render() {
    const {
      amenity,
      closeModal,
      resourceQuantity,
    } = this.state;
    return (
      <MrmModal
        title="ADD A NEW RESOURCE"
        buttonText="Add a New Resource"
        closeModal={closeModal}
        handleCloseRequest={this.handleModalStateChange}
        className="add-resource-modal"
      >
        <form className="amenity-form" onSubmit={this.handleCloseModal}>
          <div>
            <Input
              ref={this.amenity}
              id="amenity"
              name="amenity"
              placeholder="Type resource name"
              labelName="Resource Name"
              labelClass="add-resource-controls"
              value={amenity}
              onChange={this.handleInputChange}
            />
            <Input
              ref={this.resourceQuantity}
              id="resourceQuantity"
              name="resourceQuantity"
              type="number"
              placeholder="Quantity"
              labelName="Quantity"
              controlsClass="resource-filter-controls"
              labelClass="add-resource-controls"
              defaultValue={resourceQuantity}
              onChange={this.handleInputChange}
            />
            <div className="loading-btn-div">
              <ActionButtons
                withCancel
                onClickCancel={this.handleCloseModal}
                actionButtonText="ADD RESOURCE"
                onClickSubmit={this.handleCloseModal}
              />
            </div>
          </div>
        </form>
      </MrmModal>
    );
  }
}

export default AddResource;
