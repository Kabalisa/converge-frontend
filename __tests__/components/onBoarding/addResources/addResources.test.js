import React from 'react';
import { shallow } from 'enzyme';
import AddResources from '../../../../src/components/onboarding/addResources/addResources';

let component = '';

describe('ADD RESOURCES', () => {
  beforeEach(() => {
    component = shallow(<AddResources />);
  });

  it('should render properly', () => {
    expect(component).toMatchSnapshot();
  });
});
