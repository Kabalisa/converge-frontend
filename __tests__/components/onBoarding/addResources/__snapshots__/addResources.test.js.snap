// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ADD RESOURCES should render properly 1`] = `
<div>
  <OnboardingLayout
    layoutLeft={
      <div>
        <AddResourcesTitle />
        <AddRooms />
        <div>
          <UNDEFINED>
            What is the seating capacity for Cognitio?
          </UNDEFINED>
          <InputWithNumbers
            updateOuterInputs={[Function]}
          />
          <InputResources />
        </div>
      </div>
    }
    layoutRight={
      <div>
        <span>
          This part will be use to fetch data from the backend
        </span>
      </div>
    }
  />
</div>
`;
